test/Microsoft.AspNetCore.Razor.Evolution.Test/IntegrationTests/IntegrationTestBase.cs

```diff
@@ -0,0 +1,95 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.IO;
+#if NET451
+using System.Runtime.Remoting;
+using System.Runtime.Remoting.Messaging;
+#else
+using System.Threading;
+#endif
+using Microsoft.AspNetCore.Razor.Evolution.Intermediate;
+using Xunit;
+using Xunit.Sdk;
+
+namespace Microsoft.AspNetCore.Razor.Evolution.IntegrationTests
+{
+    [IntializeTestFile]
+    public abstract class IntegrationTestBase
+    {
+#if GENERATE_BASELINES
+        private static readonly bool GenerateBaselines = true;
+#else
+        private static readonly bool GenerateBaselines = false;
+#endif
+
+#if NET451
+#else
+        private static readonly AsyncLocal<string> _filename = new AsyncLocal<string>();
+#endif
+
+        // Used by the test framework to set the 'base' name for test files.
+        public static string Filename
+        {
+#if NET451
+            get
+            {
+                var handle = (ObjectHandle)CallContext.LogicalGetData("IntegrationTestBase_Filename");
+                return (string)handle.Unwrap();
+            }
+            set
+            {
+                CallContext.LogicalSetData("IntegrationTestBase_Filename", new ObjectHandle(value));
+            }
+#else
+            get { return _filename.Value; }
+            set { _filename.Value = value; }
+#endif
+        }
+
+        protected RazorCodeDocument CreateCodeDocument()
```

Comment:
`CreateIntegrationCodeDocument` or something that makes this method stick out in tests as being part of the test infrastructure.